.gitignore

```diff
@@ -105,6 +105,12 @@ cmake-build-debug/
 # IntelliJ
 out/
 
+# eclipse
```

Comment:
Sounds fine to me, but while we're at it, we should probably expand our ignores, and not use leading asterisks.

For example, here's the [Checkstyle project's .gitignore](https://github.com/checkstyle/checkstyle/blob/master/.gitignore), which maybe we should just emulate because it has more ignores that would be useful, IMHO.

Reply:
I added modifications you pointed out @sappenin.

Sideways, it seems the current ignores-list for eclipse was source from https://github.com/github/gitignore/blob/master/Global/Eclipse.gitignore . So I appended a few missing bits to that template.